src/main/drivers/bus_i2c_stm32f30x.c

```diff
@@ -231,15 +231,17 @@ bool i2cWrite(uint8_t addr_, uint8_t reg, uint8_t data)
         }
     }
 
-    /* Send Register address */
-    I2C_SendData(I2Cx, (uint8_t) reg);
+    if (reg != -1) {
```

Comment:
> (but writing single byte does not make much sense)

I can understand that impression.

The PR was originally intended to allow 0xFF as a sub-address. A minimal fix was to modify only `i2cRead`, but for a consistency of API, I decided to handle the case in `i2cWrite`, too.

The use case of the single byte write is not obvious (especially in F3 in which `i2cWriteBuffer` is absent), but compare the two calls to get an idea.
```
i2cWriteBuffer(slaveaddr, buf[0], bytes - 1, &buf[1]);
i2cWriteBuffer(slaveaddr, -1, bytes, buf);
```